src/Groups/GAPGroups.jl

```diff
@@ -0,0 +1,617 @@
+The parent of `x` is G. If `x` is not in G, it return ERROR.
+"""
+function cperm(g::PermGroup,L::Union{Array{Int64,1},UnitRange{Int64}}...)
+   if length(L)==0
+      return one(g)
+   else
+      x=GAP.Globals.Product(GAP.julia_to_gap([GAP.Globals.CycleFromList(GAP.julia_to_gap(collect(y))) for y in L]))
+      if GAP.Globals.IN(x,g.X) return PermGroupElem(g, x)
+      else throw(ArgumentError("the element does not embed in the group"))
+      end
+   end
+end
+
+"""
+    listperm(x::PermGroupElem)
+is the list L defined by L = [ `x`(i) for i in 1:n ], where Sym(n) = parent(`x`).
+"""
+function listperm(x::PermGroupElem)
+   return [x(i) for i in 1:x.parent.deg]
+end
+
+"""
+    gens(G::Group)
+return an array of generators of the group `G`.
+"""
+function gens(G::Group)
+   L = GAP.Globals.GeneratorsOfGroup(G.X)
+   res = Vector{elem_type(G)}(undef, length(L))
+   for i = 1:length(res)
+     res[i] = group_element(G, L[i])
+   end
+   return res
+end
+
+"""
+
+    gens(G::Group, i::Int)
+return the `i`-th element of the array gens(`G`). If `i` is greater than the length of gens(`G`), an ERROR is returned.
+"""
+function gens(G::Group, i::Int)
+   L = GAP.Globals.GeneratorsOfGroup(G.X)
+   @assert length(L) >= i "The number of generators is lower than the given index"
+   return group_element(G, L[i])
+end
+
+"""
+    ngens(G::Group) -> Int
+return the length of the array gens(G).
+
+!!! warning "WARNING:" 
+    this is *NOT*, in general, the minimum number of generators for G.
+"""
+ngens(G::Group) = length(GAP.Globals.GeneratorsOfGroup(G.X))
+
+
+Base.getindex(G::Group, i::Int) = gens(G, i)
+Base.sign(x::PermGroupElem) = GAP.Globals.SignPerm(x.X)
+
+Base.isless(x::PermGroupElem, y::PermGroupElem) = x<y
+
+#embedding of a permutation in permutation group
+function (G::PermGroup)(x::PermGroupElem)
+   if !GAP.Globals.IN(x.X,G.X)
+      throw(ArgumentError("the element does not embed in the group"))
+   end
+   return group_element(G, x.X)
+end
+
+#evaluation function
+function (x::PermGroupElem)(n)
+   return GAP.Globals.OnPoints(n,x.X)
+end
+
+function conjugate_subgroup(G::T, x::GAPGroupElem) where T<:Group
+  return T(GAP.Globals.ConjugateSubgroup(G.X,x.X))
+end
+
+################################################################################
+#
+#   Conjugacy Classes
+#
+################################################################################
+
+"""
+    GroupConjClass
+it could be either the conjugacy class of an element or of a subgroup in a group G. It is displayed as
+    cc = x ^ G
+where G is a group and x = `representative`(`cc`) is either an element or a subgroup of G.
+"""
+struct GroupConjClass{T<:Group, S<:Union{GAPGroupElem,Group}}
+   X::T
+   repr::S
+   CC::GapObj
+end
+
+Base.show(io::IO, x::GroupConjClass) = print(io, GAP.gap_to_julia(GAP.Globals.StringView(x.repr))*" ^ "*GAP.gap_to_julia(GAP.Globals.StringView(x.X)))
+
+function _conjugacy_class(G, g, cc::GapObj)         # function for assignment
```

Comment:
is this really necessary? It seems that a direct call to `GroupConjClass(G,g,cc)` should do (as long as there are no inner constructors)?

Reply:
It is not strictly necessary. We wrote that function because we proceeded in a similar way for other Group objects, but if we think it is superfluous, we can skip it.